cmip6_downscaling/workflows/bcsd_flow.py

```diff
@@ -0,0 +1,112 @@
+import os
+
+from dask_kubernetes import KubeCluster, make_pod_spec
+from prefect import Flow, task
+from prefect.executors import DaskExecutor
+from prefect.run_configs import KubernetesRun
+from prefect.storage import Azure
+
+from cmip6_downscaling.methods.bcsd import (
+    fit_and_predict,
+    postprocess_bcsd,
+    prep_bcsd_inputs,
+    preprocess_bcsd,
+)
+from cmip6_downscaling.workflows.utils import make_flow_paths
+
+connection_string = os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+
+run_hyperparameters = {
+    "FLOW_NAME": "BCSD_testing",
+    "GCM": "MIROC6",
+    "SCENARIO": "ssp370",
+    "TRAIN_PERIOD_START": "1990",
+    "TRAIN_PERIOD_END": "1990",
+    "PREDICT_PERIOD_START": "2090",
+    "PREDICT_PERIOD_END": "2090",
+    "VARIABLE": "tasmax",
+    "OBS": "ERA5",
+}
+
+storage = Azure("prefect")
+image = "carbonplan/cmip6-downscaling-prefect:latest"
+run_config = KubernetesRun(cpu_request=7, memory_request="16Gi", image=image, labels=["az-eu-west"])
+
+executor = DaskExecutor(
+    cluster_class=lambda: KubeCluster(
+        make_pod_spec(
+            image=image,
+            memory_limit="16Gi",
+            memory_request="16Gi",
+            threads_per_worker=2,
+            cpu_limit=2,
+            cpu_request=2,
+            env={
+                "AZURE_STORAGE_CONNECTION_STRING": os.environ["AZURE_STORAGE_CONNECTION_STRING"],
+            },
+        )
+    ),
+    adapt_kwargs={"minimum": 4, "maximum": 20},
+)
+
+
+flow_name = run_hyperparameters.pop("FLOW_NAME")  # pop it out because if you leave it in the dict
+# but don't call it as a parameter it'll complain
+
+make_flow_paths_task = task(make_flow_paths, log_stdout=True, nout=4)
```

Comment:
What is the advantage of making the imported functions task here vs in cmip6_downscaling.workflows.utils?

No idea what the preferred way, just curious on your thinking!

Reply:
I think I agree here. It seems like most of these functions should move here and just be Prefect Tasks.

Reply:
Currently the `bcsd_flow.py` script contains a prefect flow (e.g. there is a `flow.run()` call at the bottom). If the functions were specified in there we wouldn't be able to import them anywhere else without running the flow. If you know of a way to bypass that, let me know, but I recommend keeping it as-is. Especially if you might want to restructure things, as testing any changes would likely be frustrating.